test/test_nn.py

```diff
@@ -20537,6 +20537,21 @@ def test_multihead_attn_fast_path_query_and_bias_have_different_dtypes(self, dev
         query = torch.randn(3, 3, 3, dtype=dtype, device=device)
         mha(query, query, query)
 
+    @dtypes(torch.double)
+    @torch.no_grad()
+    def test_multihead_attn_in_proj_bias_none(self, device, dtype):
+        mha = torch.nn.MultiheadAttention(1, 1, bias=True, dtype=dtype, device=device)
+        query = torch.rand(3, 2, 1)
+        mha(query, query, query)
+
+    @dtypes(torch.double)
+    @torch.no_grad()
+    def test_multihead_attn_in_proj_weight_none(self, device, dtype):
+        mha = torch.nn.MultiheadAttention(4, 4, vdim=2, kdim=2, dtype=dtype, device=device)
```

Comment:
The MHA code in general is pretty hard to read :)

For clarity, I think it'd be useful to have a comment here explaining that picking kdim and vdim != embed_dim will cause the logic to use per-input projection weights, and `in_proj_weight` will be None

Reply:
Done, thanks :)